### DISCUSSION TITLE
Use of value classes to wrap Kotlin primitives

### DISCUSSION THREAD

**Discussion by pnewell** [skiptools/skip#49] [General]
In reference to this point:

> Skip does not wrap Kotlin’s primitive types. We have chosen the massive efficiency and interoperability wins that come with using Kotlin’s primitive types directly over the additional Swift language compatibility we might be able to achieve if we wrapped Kotlin’s primitives in our own classes.

Have you all considered using Kotlin [inline value classes](https://kotlinlang.org/docs/inline-classes.html) to wrap these? My understanding is that performance is largely the same as using the primitive itself.

This could unlock protocols for primitives and could also help strings in particular to act in a much more Swift-like way.

**Comment by marcprux** [comment]
We've discussed it in the past, and we're open to further discussion. The advantages you mention are enticing.
